tests/neon/test_neon_wasm_simd.cpp

```diff
@@ -0,0 +1,290 @@
+                        { { 208, 154, 88, 53, 92, 83, 7, 151,
+                            86, 2, 140, 126, 113, 126, 50, 125 },
+                          { -41,  3, -33, -68,  95, -51, -63, -32,
+                            -67,  5,  69, -45,  88, -114, -125,  41 },
+                          {  40, -37,  94, -124,  46,  101,  27, -124,
+                             103, -89,  2, -40,  38,  52,  85, -3 },
+                          { -8,  67,  94, -76,  126,  101,  25,      INT8_MIN,
+                            53, -91,  6, -46,  86,  14,  71, -87 } },
+                        { { 55, 52, 186, 150, 1, 123, 119, 190,
+                            128, 188, 145, 217, 74, 21, 2, 115 },
+                          { -16,  96, -9,  31, -58,  19, -93,  45,
+                            -70, -90,  6, -32, -38,  91, -34,  18 },
+                          { -112, -104, -88, -111,  19,  31,  80, -108,
+                            -36, -31,  109,  38, -10,  111, -103, -25 },
+                          { -80, -88, -78,  23,  18,  23,  35,  44,
+                            -36, -27,  108, -26, -2,  123, -101, -106 } },
+        };
+
+        for (size_t i = 0 ; i < (sizeof(test_vec) / sizeof(test_vec[0])) ; i++) {
+                uint8x16_t a = vld1q_u8(test_vec[i].a);
+                int8x16_t b = vld1q_s8(test_vec[i].b);
+                int8x16_t c = vld1q_s8(test_vec[i].c);
+                int8x16_t r = vbslq_s8(a, b, c);
+
+                int8_t r_[16];
+                vst1q_s8(r_, r);
+                assert(memcmp(r_, test_vec[i].r, sizeof(int8_t) * 16) == 0);
+
+        }
+
+        return 0;
+}
+
+static int
+test_simde_vshl_s64 () {
+  struct {
+    int64_t a[1];
+    int64_t b[1];
+    int64_t r[1];
+  } test_vec[] = {
+    { { -4131760704340726406 },
+      { -51 },
+      { -1835 } },
+    { {  4795347804945835666 },
+      { -8617733424538507080 },
+      {  0 } },
+    { { -2987477723010756817 },
+      { -49 },
+      { -5307 } },
+    { { -834069088546172233 },
+      {  62 },
+      { -4611686018427387904 } },
+    { { -3249651920531950297 },
+      {  2418075451758470935 },
+      {  7338760155284111360 } },
+    { {  5243402359250859729 },
+      { -47 },
+      {  37256 } },
+    { { -5953442574285360237 },
+      {  33 },
+      { -1003907829020491776 } },
+    { {  7218011237552599699 },
+      { -36 },
+      {  105035887 } },
+  };
+
+  for (size_t i = 0 ; i < (sizeof(test_vec) / sizeof(test_vec[0])) ; i++) {
+    int64x1_t a = vld1_s64(test_vec[i].a);
+    int64x1_t b = vld1_s64(test_vec[i].b);
+    int64x1_t r = vshl_s64(a, b);
+    int64_t r_[1];
+    vst1_s64(r_, r);
+    assert(memcmp(r_, test_vec[i].r, sizeof(int64_t) * 1) == 0);
+  }
+
+  return 0;
+}
+
+int main() {
+        printf("Testing NEON Wasm SIMD\n");
+        test_simde_vaddq_s32();
+        test_simde_vsubq_s16();
+        test_simde_vmulq_u32();
+        test_simde_vbslq_s8();
+        test_simde_vshl_s64();
+        printf("Success!\n");
+}
```

Comment:
Good to have a smoke test, though this kind of manually written test will likely not scale. Would be good to have an autogenerated native vs web test like x86 SSE tests have, that would test the full API. That could be run on e.g. a Raspberry Pi 4, or on the upcoming MacBooks, and skip otherwise if host system is not ARM NEON capable?

Reply:
This smoke test is just to make sure the header can be imported correctly. The implementation itself has comprehensive tests upstream in SIMDe that we won't need to duplicate here.